src/controllers/auth.controller.js

```diff
@@ -0,0 +1,189 @@
+import { User } from '../models/user.js';
+import { userService } from '../services/user.service.js';
+import { jwtService } from '../services/jwt.js';
+import { ApiError } from '../exception/api.error.js';
+import bcrypt from 'bcrypt';
+import { tokenService } from '../services/token.service.js';
+
+function validateName(value) {
+  if (!value) {
+    return 'Name is required';
+  }
+
+  if (value.includes(' ')) {
+    return 'One word required';
+  }
+
+  if (value.length < 2) {
+    return 'At least 2 characters';
+  }
+}
+
+function validateEmail(value) {
+  if (!value) {
+    return 'Email is required';
+  }
+
+  const emailPattern = /^[\w.+-]+@([\w-]+\.){1,3}[\w-]{2,}$/;
+
+  if (!emailPattern.test(value)) {
+    return 'Email is not valid';
+  }
+}
+
+const validatePassword = (value) => {
+  if (!value) {
+    return 'Password is required';
+  }
+
+  if (value.length < 6) {
+    return 'At least 6 characters';
+  }
+};
+
+const registration = async (req, res, next) => {
+  const { name, email, password } = req.body;
+
+  const errors = {
+    name: validateName(name),
+    email: validateEmail(email),
+    password: validatePassword(password),
+  };
+
+  if (errors.name || errors.email || errors.password) {
+    throw ApiError.badRequest('Bad request', errors);
+  }
+
+  const hashedPass = await bcrypt.hash(password, 5);
+
+  await userService.register(name, email, hashedPass);
+
+  res.send({ message: 'OK' });
+};
+
+const changePassword = async (req, res) => {
+  const { password, confirmation } = req.body;
+  const { activationToken } = req.params;
+
+  if (password !== confirmation) {
+    throw ApiError.badRequest('Values should match');
+  }
+
+  if (validatePassword(password)) {
+    throw ApiError.badRequest('Password should be 6 characters at least');
```

Comment:
The condition here is incorrect. `validatePassword` returns an error message if the password is invalid, so you should check if the result is truthy instead of checking for a specific condition.